### ISSUE TITLE
Error 250 on Zoho

### ISSUE THREAD

**Issue by Zen0x7** [jeremydumais/CPP-SMTPClient-library#28]
Hi @jeremydumais 

I recently move my emails between providers, from google to zoho but now i have a issue, can't send emails, logs shared:

```
c: Trying to connect to smtppro.zoho.com on port 587
s: 220 mx.zohomail.com SMTP Server ready April 18, 2024 6:38:02 AM PDT
c: Connected!
c: ehlo localhost\r\n
s: 250-mx.zohomail.com Hello localhost ({HIDDEN} ({HIDDEN}))
250-STARTTLS
c: Info: STARTTLS is available by the server, the communication will be encrypted.
c: STARTTLS\r\n
s: 250 SIZE 53477376
An error occurred: No message correspond to this error code (error no: 250)
```

Using a smtp tool to check if server is able to send emails, got the following log:

```
<< 220 mx.zohomail.com SMTP Server ready April 18, 2024 6:40:27 AM PDT
>> EHLO [172.31.11.248]
<< 250-mx.zohomail.com Hello [{HIDDEN}] (ec2-54-212-131-181.us-west-2.compute.amazonaws.com (54.212.131.181))
<< 250-STARTTLS
<< 250 SIZE 53477376
>> STARTTLS
<< 220 Ready to start TLS.
>> EHLO [172.31.11.248]
<< 250-mx.zohomail.com Hello [{HIDDEN}] (ec2-54-212-131-181.us-west-2.compute.amazonaws.com (54.212.131.181))
<< 250-AUTH LOGIN PLAIN
<< 250 SIZE 53477376
>> AUTH PLAIN =====PLAIN_HASH======
<< 235 Authentication Successful
>> MAIL FROM:<{HIDDEN}> SIZE=550
<< 250 Sender <{HIDDEN}> OK
>> RCPT TO:<{HIDDEN}>
<< 250 Recipient <{HIDDEN}> OK
>> DATA
<< 354 Ok Send data ending with <CRLF>.<CRLF>
>> From: {HIDDEN}
>> Date: Thu, 18 Apr 2024 13:40:17 퍍
>> Subject: SMTP test from smtppro.zoho.com
>> Message-Id: <{HIDDEN}@{HIDDEN}>
>> To: ian@zentrack.cl
>> MIME-Version: 1.0
>> Content-Type: multipart/alternative; boundary="={HIDDEN}=="
>>
>> --=-{HIDDEN}==
>> Content-Type: text/plain; charset=utf-8
>>
>> Test message
>> --={HIDDEN}==
>> Content-Type: text/html; charset=utf-8
>> Content-Id: <{HIDDEN}>
>>
>> <b>Test message</b>
>> --={HIDDEN}==--
>> .
<< 250 Message received
```

Could you check if something went wrong? Because tool used sends STARTTLS after 250 SIZE message.

**Comment by jeremydumais** [comment]
Hi @Zen0x7,

Ok after investigation, it seems that the Zoho SMTP server is returning the list of the extensions it supports on different buffers.

It returns 
50-mx.zohomail.com Hello localhost ({HIDDEN} ({HIDDEN}))
250-STARTTLS

...and then when I read again later it return the remaining extension
250 SIZE 53477376

This is a different behavior than smtp.live.com and smtp.gmail.com.

I will look to provide a fix that this support scenario.

**Comment by jeremydumais** [comment]
This scenario is fixed in version 1.1.8 that will be release later today.

Thank you,
